TappNetwork/home-visiting-app-web
# Cu 25tmceq internal referrals

# Details

This PR encompasses changes related to the internal referrals features  for both the front and back end of the Home Visiting App. Visitors are now able to refer patients to other visitors with the following steps:

1.  Click the referral button on Visitor B's profile to open a modal where the Visitor A will craft a message referring their patient to Visitor B.
2. Visitor B will see the referral on their dashboard or in their referral inbox. Visitor B will be able to then accept or reject the referral. When Visitor B opens the referral, Visitor A will be able to see that the referral has been seen, even if Visitor B does not act on the referral.
3. Visitor A can see the response from Visitor B in their inbox as well as contact information if the referral was accepted

# Issues

1. The vue router issue from the previous PR has not been resolved as of yet, and also happens if user clicks a sidebar link while modal is open
2. Tests for the backend run veeeerrry slow.

# Steps to test changes

1. Pull this branch in front and back end
3. run 'npm run serve' in front end repo
4. run 'php artisan migrate:fresh --seed' in backend repo
5. run 'php artisan test' in backend repo and ensure all tests pass
6. go to localhost:8080 in browser
7. go to visitor directory and select a visitor profile
8. click 'make referral' button and complete modal
9. go to referral inbox and ensure new referral is present
10. test all filters and search functionality
11. respond to referral awaiting your request and ensure status or referral changes
12. go to dashboard and respond to one of the listed referrals, check inbox to see status changed
13. Test seen functionality by clicking on completed referral, clicking it again and seeing that the seen status is displayed

# Issues Addressed

[25tmceq](https://app.clickup.com/t/25tmceq)
